controllers/applicationvolumereplication_controller.go

```diff
@@ -417,9 +420,53 @@ func (a *AVRInstance) processPausedSubscription(
 		return IsManifestInAppliedState(pvMW)
 	}
 
+	ready := a.reconciler.isManagedClusterReadyForFailback(subscription, newHomeCluster)
+	a.reconciler.Log.Info(newHomeCluster, "is ready for failback:", ready)
+
+	if !ready {
+		return !unpause
+	}
+
 	return a.cleanupAndRestore(subscription, newHomeCluster)
 }
 
+func (r *ApplicationVolumeReplicationReconciler) isManagedClusterReadyForFailback(
```

Comment:
The function name should be changed.  Maybe something like getVRGFromManagedCluster() that returns a VRG object.  Then let the caller do whatever it needs to do with the VRG object.

Reply:
Sounds reasonable. Implemented this on 6338994c1484bc4ee8bbdc8a2ea19d05ca5d2a1f